org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/repositories/actions/OpenRepositoryAction.java

```diff
@@ -36,20 +43,50 @@ public OpenRepositoryAction(IViewPart view) {
 
 	@Override
 	public void run() {
+
 		IRepository repository = getRepository();
 		if (repository != null) {
 			try {
-				// Open default external browser
-				PlatformUI.getWorkbench().getBrowserSupport().getExternalBrowser().openURL(new URL(repository.getUrl()));
+				// Get Repository link
+				String repoLink = getLink(repository);
+				// Open the link in default external browser
+				URI uri = new URI(repoLink);
+				PlatformUI.getWorkbench().getBrowserSupport().getExternalBrowser().openURL(uri.toURL());
 			} catch (PartInitException | MalformedURLException exception) {
 				MessageDialog.openError(this.view.getViewSite().getShell(), Messages.AbapGitView_action_open_repo_error_dialog_title,
 						exception.getMessage());
 				AbapGitUIPlugin.getDefault().getLog()
 						.log(new Status(IStatus.ERROR, AbapGitUIPlugin.PLUGIN_ID, exception.getMessage(), exception));
+			} catch (URISyntaxException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
 			}
 		}
 	}
 
+	public String getLink(IRepository repository) {
+		String repoLink = repository.getUrl();
+		repoLink = repoLink.replace(".git", ""); //$NON-NLS-1$ //$NON-NLS-2$
+		// Remove protocol (http:// or https://)
+		String domain = repoLink.replaceFirst("https?://", ""); //$NON-NLS-1$ //$NON-NLS-2$
+
+		// Extract domain by splitting with '/'
+		domain = domain.split("/")[0]; //$NON-NLS-1$
+		// Get connected branch
+		String branch = repository.getBranchName();
+		// handle domain based repoLink generation
+		if (domain.equals(GITHUB_DOMAIN) || domain.equals(GITLAB_DOMAIN) || domain.equals(GITHUB_TOOLS_DOMAIN)
+				|| domain.equals(GITHUB_INFRA_DOMAIN) || domain.equals(GIT_WDF_SAP_DOMAIN)) {
+			branch = branch.replace("refs/heads/", "/tree/"); //$NON-NLS-1$//$NON-NLS-2$
+		} else if (domain.endsWith(BIT_BUCKET_DOMAIN)) {
+			repoLink = repoLink.replaceAll("https://.*?@", "https://"); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
You can also remove the username in BitBucket links by using URI to parse the repoLink.
Then rebuild the URI without userInfo & get URL.

Check constructor for URI:
  public URI(String scheme,
               String userInfo, String host, int port,
               String path, String query, String fragment)

Reply:
Also check if other GIT providers can have user info.
In such case, we could have a domain agnostic user info handler.

Reply:
Didn't get any user info in case of git providers.